client/src/components/pages/TicketsPage/index.jsx

```diff
@@ -0,0 +1,102 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import MainSidebar from '../../parts/MainSidebar';
+import TicketsSidebar from '../../parts/TicketsSidebar';
+import Navbar from '../../parts/Navbar';
+import WrappedTicket from '../../parts/WrappedTicket';
+import './style.css';
+
+export default class extends Component {
+  state = {
+    tickets: this.props.tickets || {
+      all: {
+        pending: [],
+        closed: [],
+      },
+      my: {
+        pending: [],
+        closed: [],
+      },
+      drafts: [],
+      trash: [],
+    },
+    currentCategory: this.props.match.params.category || 'all',
+    currentStatus: this.props.match.params.status || 'pending',
+    allChecked: false,
+    markAs: null,
+  };
+
+  componentWillReceiveProps(nextProps) {
+    this.setState({
+      tickets: nextProps.tickets,
+      currentCategory: nextProps.match.params.category,
+      currentStatus: nextProps.match.params.status,
+    });
+  }
+
+  toggleAllChecked = () => {
+    this.setState({ allChecked: !this.state.allChecked });
+  };
+
+  render() {
+    const { currentCategory, currentStatus, allChecked, markAs, tickets } = this.state;
+
+    if (!(currentCategory === 'trash' || currentCategory === 'drafts')) {
+      var pendingLength = tickets[currentCategory].pending.length,
```

Comment:
It's more descriptive to replace the ```Length``` with ```ticketCount```.
```suggestion
      var pendingTicketCount= tickets[currentCategory].pending.length,
      allTicketCount = tickets.all.pending.length + tickets.all.closed.length,
```

and so on.